wkcuber/metadata.py

```diff
@@ -86,7 +86,13 @@ def detect_dtype(dataset_path, layer, mag=1):
     layer_path = path.join(dataset_path, layer, str(mag))
     if path.exists(layer_path):
         with wkw.Dataset.open(layer_path) as dataset:
-            return str(np.dtype(dataset.header.voxel_type))
+            voxel_type = dataset.header.voxel_type
+            num_channels = dataset.header.num_channels
+            voxel_size = np.dtype(voxel_type)
+            if voxel_size == np.uint8 and num_channels > 1:
+                return "uint" + str(8 * num_channels)
```

Comment:
@normanrz The code for writing the element class into the meta data is quite ugly. Do you have a better idea here? WK expects "uint24" for multi-channel uint8 data. However, numpy doesn't even know uint24 as a data type. Maybe we should change WK's interpretation instead (the meta data could include voxel type and channel count independent of each other? However, backwards compatibility would break then.).

Reply:
I wouldn't want to change the wk interpretation. This metadata is redundant anyways, because the information is also stored in the wkw header.